packages/react-async/src/useAsync.tsx

```diff
@@ -296,20 +287,22 @@ function useAsyncFetch<T>(
     globalScope.fetch(input, init).then(parseResponse(accept, json))
   const isDefer =
     typeof defer === "boolean" ? defer : ["POST", "PUT", "PATCH", "DELETE"].indexOf(method!) !== -1
-  const fn = isDefer ? "deferFn" : "promiseFn"
+
   const identity = JSON.stringify({
     resource,
     init,
     isDefer,
   })
+
   const promiseFn = useCallback(
-    (_: AsyncOptions<T>, { signal }: AbortController) => {
+    (context: C, _: AsyncOptions<T, C>, { signal }: AbortController) => {
```

Comment:
Since the 2nd parameter's name starts with an underscore to indicate it isn't used, the 1st one should do the same for consistency/clarity.

```suggestion
    (_context: C, _options: AsyncOptions<T, C>, { signal }: AbortController) => {
```

Also, my other comment about `C` and `RequestInit` being the same thing applies here as well.